### PR TITLE
Refactor CesiumGltfComponent.cpp

### PR DESCRIPTION
**Pull request by nithinp7**
Attempts to slightly clean up the hierarchical glTF loading in `CesiumGltfComponent.cpp`. The main goal is to make gltf loading more easily extensible as we grow support for extensions. Hopefully this means future changes / additions to the gltf loading process can be more localized to only the functionality and hierarchy level that they effect.

Main Changes:
- `CreateModelOptions` replaced by several structs for model options, node options, mesh options, and primitive options. Each struct can be extended easily in the future to contain new options (usually from glTF extensions) at that level. For example, nodes usually have a single mesh, but with the MAXAR_mesh_variants extension, an arbitrary number of meshes may be associated with a node. Similarly, in EXT_mesh_gpu_instancing, the node-level extension can indicate that its mesh (and so all the primitives coming from the mesh as well) should be instanced. These are examples of what could eventually go inside some of these options struct, even though most of them are empty for now.
- `LoadModelResult` replaced by several structs (model result, node result, mesh result, and primitive result) to retain the hierarchy of the result of the loading. Similar to the above change, this will help new additions be localized to a specific level (node, mesh, primitive, etc) and the result of that loading will be discernible conveniently at that same level.
- Changed some function / struct names that were misleading about what they were doing. E.g. `LoadModelResult` has been changed to `LoadPrimitiveResult`, `loadModelGameThreadPart` has been changed to `loadPrimitiveGameThreadPart`.
- ... hopefully more!

CC: @kring @baothientran @joseph-kaile 
Do you folks have any suggestions / objections for refactoring this code and how I'm going about it here? I'm open to more suggestions for other things to clean up / refactor in this PR related to the gltf loading.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @nithinp7!

* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium-unreal/blob/gltf-refactor/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] [Cesium for Unreal Samples](https://github.com/CesiumGS/cesium-unreal-samples) works.

File: Source/CesiumRuntime/Private/CesiumGltfComponent.cpp [hunk 1]
```diff
@@ -1357,21 +1322,31 @@ static std::vector<LoadModelResult> loadModelAnyThreadPart(
     // Show the default scene
     const CesiumGltf::Scene& defaultScene = model.scenes[model.scene];
     for (int nodeId : defaultScene.nodes) {
-      loadNode(result, model, model.nodes[nodeId], rootTransform, options);
+      CreateNodeOptions nodeOptions = options;
+      nodeOptions.pNode = &model.nodes[nodeId];
```
Comment (kring) [inline Source/CesiumRuntime/Private/CesiumGltfComponent.cpp:1326]: I think it'd be a little clearer and safer to construct the CreateNodeOptions from the two things explicitly, like:

```
CreateNodeOptions nodeOptions(options, &model.nodes[nodeId]);
```

**Review by kring — COMMENTED** [review]
This looks good to me @nithinp7. I only have one nitpick below, and it doesn't even need to hold up merging this. So just move it out of the draft state and I'll merge it.

**Comment by kring** [comment]
👍

**Pull request merged**